pkg/storage/wal/manager.go

```diff
@@ -107,38 +105,50 @@ func NewMetrics(r prometheus.Registerer) *Metrics {
-// available and which are waiting to be flushed. The maximum number of
-// segments that can be buffered in memory, and their maximum age and maximum
-// size before being flushed are configured when creating the Manager.
+// accepting writes and which are waiting to be flushed. It uses two doubly
+// linked lists called the available and pending lists, and as such a Manager
+// has constant time complexity.
+//
+// The maximum number of segments that can be buffered in memory, and their
+// maximum age and maximum size before being flushed are configured when
+// creating a Manager. When a segment exceeds the maximum age or maximum size
+// it is moved to the pending list.
 //
-// By buffering segments in memory, the WAL can tolerate bursts of append
-// requests that arrive faster than can be flushed. The amount of data that can
-// be buffered is configured using MaxSegments and MaxSegmentSize. You must use
-// caution when configuring these to avoid excessive latency.
+// By buffering segments in memory, the WAL can tolerate bursts of writes that
+// arrive faster than can be flushed. The amount of data that can be buffered
+// is configured using MaxSegments and MaxSegmentSize. You must use caution
+// when configuring these to avoid excessive latency.
 //
-// The WAL is full when all segments are waiting to be flushed or in the process
-// of being flushed. When the WAL is full, subsequent appends fail with ErrFull.
-// It is not permitted to append more data until another segment has been flushed
-// and returned to the available list. This allows the manager to apply back-pressure
-// and avoid congestion collapse due to excessive timeouts and retries.
+// The WAL is full when all segments are waiting to be flushed or in the
+// process of being flushed. When the WAL is full, subsequent appends fail with
+// ErrFull. It is not permitted to append more data until another segment has
+// been flushed and returned to the available list. This allows the manager to
+// apply back-pressure to clients that may be timing out and retrying.
 type Manager struct {
 	cfg     Config
 	metrics *Metrics
 
-	// available is a list of segments that are available and accepting data.
-	// All segments other than the segment at the front of the list are empty,
-	// and only the segment at the front of the list is written to. When this
-	// segment has exceeded its maximum age or maximum size it is moved to the
-	// pending list to be flushed, and the next segment in the available list
-	// takes its place.
+	// available is a list of segments that are accepting writes. All segments
+	// other than the segment at the front of the list are empty, and only the
+	// segment at the front of the list is written to. When this segment has
+	// exceeded its maximum age or maximum size it is moved to the pending list
+	// to be flushed, and the next segment in the available list takes its place.
 	available *list.List
```

Comment:
The comment could be much simpler here with a name that was a bit more descriptive, `availableSegments` or `availableWriteSegments`?

Reply:
I'm not convinced making the variable name more verbose will add value here? There are some good comments above that explains what the manager does and how it works:

> // Manager buffers segments in memory, and keeps track of which segments are
> // accepting writes and which are waiting to be flushed. It uses two doubly
> // linked lists called the available and pending lists, and as such a Manager
> // has constant time complexity.